### PR TITLE
Initial rework for in-notebook editor

### PR DESCRIPTION
**Pull request by pentacular**

### CODE REVIEW

File: api/shape/Curve.js [hunk 1]
```diff
@@ -8,17 +8,24 @@ export const Curve = Shape.registerMethod(
   'Curve',
```
Comment (BarbourSmith) [inline api/shape/Curve.js:8]: What do curves mean?

File: ui/app/EditNote.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import * as PropTypes from 'prop-types';
+
+import { CodeJar } from 'codejar';
+import React from 'react';
+
+export class EditNote extends React.Component {
```
Comment (BarbourSmith) [inline ui/app/EditNote.js:6]: I guess the react stuff is paying off

File: nb/documentation/creating_geometry/creating_geometry.md [hunk 3]
```diff
@@ -74,57 +74,59 @@ Empty().view();
...
 Links a set of ordered points together into ordered segments.
 
+![Image](creating_geometry.md.$20.png)
+
 ```JavaScript
 await Group(Point(0, 0), Point(10, 10), Point(20, 10)).link().view();
 ```
 
-![Image](creating_geometry.md.$20.png)
+![Image](creating_geometry.md.$21.png)
```
Comment (BarbourSmith) [inline nb/documentation/creating_geometry/creating_geometry.md:129]: These are all automatically numbered, right? You didn't have to change all of these?

**Review by BarbourSmith — APPROVED** [review]
Since this PR touches on so many things it was a nice refresher :)

It's impressive how compact the actual code changes are to do the in-notebook editing.

Is it up on /alpha to play with?

Looks great to me!

File: nb/documentation/creating_geometry/creating_geometry.md (hunk 3 above)
Comment (pentacular) [reply nb/documentation/creating_geometry/creating_geometry.md:129]: Yes :)

File: api/shape/Curve.js (hunk 1 above)
Comment (pentacular) [reply api/shape/Curve.js:8]: It produces a curved path that passes smoothly through the control points.

**Comment by pentacular** [comment]
> Since this PR touches on so many things it was a nice refresher :)
> 
> It's impressive how compact the actual code changes are to do the in-notebook editing.
> 
> Is it up on /alpha to play with?
> 
> Looks great to me!

I'll put it up after merging.

It's a bit on the fragile side, so things may break a bit.

**Pull request merged**